st/clitests/awsiam_spec.py

```diff
@@ -191,7 +201,148 @@ def policy_tests():
+        quit()
+    result.command_response_should_have("iampolicy")
+
+    #list user attached policies with path prefix filter
+    result = AwsIamTest('List User Attached Policies').list_attached_user_policies("testUser", \
+        path_prefix="/listpolicy/").execute_test()
+    total_policies = get_attached_user_policy_list_count(result.status.stdout, "ATTACHEDPOLICIES")
+    if(total_policies != 1):
+        print('List Policies Test failed')
+        quit()
+    result.command_response_should_have("iampolicy2")
+
+	#list user policies for user not having any policies attached
+    result = AwsIamTest('List User Attached Policies').list_attached_user_policies("testUser2") \
+        .execute_test()
+    total_policies = get_attached_user_policy_list_count(result.status.stdout, "ATTACHEDPOLICIES")
+    if(total_policies != 0):
+        print('List Policies Test failed')
+        quit()
+
+    #list user policies for wrong username
+    AwsIamTest('List User Attached Policies').list_attached_user_policies("asdhj") \
+        .execute_test(negative_case=True).command_should_fail().command_error_should_have("NoSuchEntity")
+
+	#detach-user-policy
+    AwsIamTest('Detach User Policy').detach_user_policy("testUser", arn).execute_test() \
+        .command_is_successful()
+    AwsIamTest('Detach User Policy').detach_user_policy("testUser", arn2).execute_test() \
+        .command_is_successful()
+
+    #detach-user-policy for non existing user
+    result = AwsIamTest('Detach User Policy').detach_user_policy("hkhk", arn) \
+        .execute_test(negative_case=True).command_should_fail().command_error_should_have("NoSuchEntity")
+
+    #detach-user-policy with non existing policy
+    result = AwsIamTest('Detach User Policy').detach_user_policy("testUser", "arn:seagate:iam::103947660857:policy/oiuou") \
+        .execute_test(negative_case=True).command_should_fail().command_error_should_have("NoSuchEntity")
+
+    #detach-user-policy with non attached policy
+    result = AwsIamTest('Detach User Policy').detach_user_policy("testUser", arn) \
+        .execute_test(negative_case=True).command_should_fail().command_error_should_have("NoSuchEntity")
+
+	#list user policies after detaching the policies
+    result = AwsIamTest('List User Attached Policies').list_attached_user_policies("testUser").execute_test()
+    total_policies = get_attached_user_policy_list_count(result.status.stdout, "ATTACHEDPOLICIES")
+    if(total_policies != 0):
+        print('List Policies Test failed')
+        quit()
+
+    #delete-policy
+    result = AwsIamTest('Delete Policy').delete_policy(arn).execute_test().command_is_successful()
+    result = AwsIamTest('Delete Policy').delete_policy(arn2).execute_test().command_is_successful()
+
+    policy_arns = []
+    non_attached_policy_arns = []
+    for i in range(21):
+        #create-policy
+        samplepolicy = os.path.join(os.path.dirname(__file__), 'policy_files', 'iam-policy.json')
```

Comment:
Can we shift the logic in separate method-eg - populateArns (..)

Reply:
get_arn_from_policy_object() function is already present to get the arn from the command result. So I don't think there needs a separate method again.

Reply:
Its better to separate it out as here you just preparing data for ST

Reply:
As per discussion @shalakadharap  will be handling this in PR https://github.com/Seagate/cortx-s3server/pull/1610